File: swift/tests/WalletKitTests/FlamingoMatcherTest.swift
import Foundation
import XCTest
@testable import WalletKit

final class FlamingoMatcherTest: XCTestCase {
    func testFluentConfiguration() throws {
        let matcher = try FlamingoMatcher(hostUrl: "https://verifier.example.com")
            .withMeasurements(measurements: [
                0: Data(repeating: 1, count: 48),
                1: Data(repeating: 2, count: 48),
                2: Data(repeating: 3, count: 48)
            ])
            .withHeaders(headers: ["Authorization": "Bearer test-token"])

        XCTAssertNoThrow(try matcher.withHeaders(headers: [:]))
        XCTAssertThrowsError(try matcher.withHeaders(headers: ["x-test": "invalid\nvalue"]))
        XCTAssertThrowsError(try matcher.withMeasurements(measurements: [:]))
        XCTAssertThrowsError(try matcher.withMeasurements(measurements: [
            0: Data(repeating: 0, count: 48),
            1: Data(repeating: 2, count: 48),
            2: Data(repeating: 3, count: 48)
        ]))
    }
}

File: kotlin/walletkit-tests/src/test/kotlin/org/world/walletkit/FlamingoMatcherTest.kt
package org.world.walletkit

import uniffi.walletkit_core.FlamingoMatcher
import kotlin.test.Test
import kotlin.test.assertFails

class FlamingoMatcherTest {
    @Test
    fun fluentConfiguration() {
        val matcher =
            FlamingoMatcher("https://verifier.example.com")
                .withMeasurements(
                    mapOf(
                        0u to ByteArray(48) { 1 },
                        1u to ByteArray(48) { 2 },
                        2u to ByteArray(48) { 3 },
                    ),
                ).withHeaders(mapOf("Authorization" to "Bearer test-token"))

        matcher.withHeaders(emptyMap())
        assertFails { matcher.withHeaders(mapOf("x-test" to "invalid\nvalue")) }
        assertFails { matcher.withMeasurements(emptyMap()) }
        assertFails {
            matcher.withMeasurements(
                mapOf(
                    0u to ByteArray(48),
                    1u to ByteArray(48) { 2 },
                    2u to ByteArray(48) { 3 },
                ),
            )
        }
    }
}
